### DISCUSSION TITLE
How to access Minio-(Operator) (Kubernetes) without UI and without NodePort?

### DISCUSSION THREAD

**Discussion by BigFlagBurito** [minio/minio#20064] [Q&A]
Hello everyone,

I've reached a point where I just don't know what to do and the instructions don't help either.
MinIO is very new to me and I have to set it up in Kubernetes. My goal is to store backups via MinIO later.
Deploying it with Helm was no problem, but then the problems started.
The official instructions indicate that you can access MinIO if you use a NordePort or port forwarding, but both should only be used for "testing purposes". For security reasons I cannot use this.
The last method would be to use an Ingress, on the one hand the UI is probably [deprecated for years](https://github.com/minio/operator/pull/2205) and on the other hand the Ingress does not work if a subpath is used.

My other idea was to use MinIO-Client.
But I don't know whether the client should connect to the console or the operator and I get error messages when I try to use both.
`mc alias set minio http://console.minio-operator.svc.cluster.local:9090 minioadmin minioadmin`
Error message:
`mc: <ERROR> Unable to initialize new alias from the provided credentials. XML syntax error on line 1: element <link> closed by </head>.`

`mc alias set minio http://operator.minio-operator.svc.cluster.local:4221 minioadmin minioadmin`
This was successful, but the connection test fails.
`mc admin info minio`
Error message:
`mc: <ERROR> Unable to get service info. Failed to parse server response (invalid character 'p' after top-level value): 404 page not found.`

I hope one of you can help me.
How do I successfully set up MinIO in Kubernetes so that I can later make backups of Kubernetes with Velero.
It should be enough if I only have one bucket.
Many thanks in advance.

**Comment by marktheunissen** [comment]
Hi @BigFlagBurito, it looks like you're trying to run before you can crawl, if I could use that analogy. 

Start by playing around with MinIO locally, and get an understanding of how the S3 API port, and the Console port work and how they might be useful for you, and read the MinIO docs to learn how it works. The basics will help.

You don't need to connect to the operator console, indeed it has been removed.
You probably just want to expose the API (S3) port to the service that needs it.

With ingress (nodeport vs load balancer) it totally depends on your Kubernetes environment and how you need to connect your service to MinIO. 

`mc` must connect to MinIO's S3 port (the API port), not the operator, and not the MinIO console. You can test this locally and with play.min.io before you start trying to get Kubernetes to work
